### PR TITLE
types: warning information is inconsistent with MySQL when convert string to double/float

### LINKED ISSUE #24888: types: warning information is inconsistent with MySQL when convert string to double/float
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Do the following in MySQL and TiDB
```
select '1x' + 1;
show warnings
```

### 2. What did you expect to see? (Required)

The warning content of both should be
```
mysql> show warnings;
+---------+------+----------------------------------------+
| Level   | Code | Message                                |
+---------+------+----------------------------------------+
| Warning | 1292 | Truncated incorrect DOUBLE value: '1x' |
+---------+------+----------------------------------------+
```
### 3. What did you see instead (Required)
MySQL:
```
+---------+------+----------------------------------------+
| Level   | Code | Message                                |
+---------+------+----------------------------------------+
| Warning | 1292 | Truncated incorrect DOUBLE value: '1x' |
+---------+------+----------------------------------------+
```

TiDB:
```
+---------+------+----------------------------------------+
| Level   | Code | Message                                |
+---------+------+----------------------------------------+
| Warning | 1292 | Truncated incorrect FLOAT value: '1x' |
+---------+------+----------------------------------------+
```

### 4. What is your TiDB version? (Required)

Master branch

### PR DESCRIPTION
**Pull request by yuqi1129**
### What problem does this PR solve?

Issue Number: close #24888 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
Minor change, see code
How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

-  `No release note`

### CODE REVIEW

**Comment by yuqi1129** [comment]
@XuHuaiyu Can you take time to view this PR, this is a small chanage

File: executor/show_test.go [hunk 1]
```diff
@@ -104,9 +104,9 @@ func (s *testSuite5) TestShowWarnings(c *C) {
 	tk.MustExec("set @@sql_mode=''")
 	tk.MustExec("insert show_warnings values ('a')")
 	c.Assert(tk.Se.GetSessionVars().StmtCtx.WarningCount(), Equals, uint16(1))
-	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1292|Truncated incorrect FLOAT value: 'a'"))
```
Comment (tangenta) [inline executor/show_test.go:107]: I found out that this is not compatible with MySQL either...

```sql
drop table if exists t;
set @@sql_mode='';
create table t (a int);
insert into t values ('a');
show warnings;
```

```
+---------+------+------------------------------------------------------+
| Level   | Code | Message                                              |
+---------+------+------------------------------------------------------+
| Warning | 1366 | Incorrect integer value: 'a' for column 'a' at row 1 |
+---------+------+------------------------------------------------------+
```

Comment (yuqi1129) [reply executor/show_test.go:107]: @tangenta Yes, you are right,  insert sentence do have different warning/error information, i have already create a new issue, see #24927

**Review by tangenta — APPROVED** [review]

**Comment by yuqi1129** [comment]
Can anyone help to review this PR ?

**Review by guo-shaoge — APPROVED** [review]

**Pull request merged**